libs/search/src/lib/results-layout/results-layout.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core'
+import { select, Store } from '@ngrx/store'
+import { UpdateResultsLayout } from '../state/actions'
+import { SearchState } from '../state/reducer'
+import { getSearchResultsLayout } from '../state/selectors'
+import { ResultsListLayout } from '@lib/common'
+
+@Component({
+  selector: 'search-results-layout',
+  templateUrl: './results-layout.component.html',
+})
+export class ResultsLayoutComponent implements OnInit {
+  choices = Object.values(ResultsListLayout).map((v) => {
+    return {
+      label: v,
+      value: v,
```

Comment:
Like in the sort by https://github.com/geonetwork/geonetwork-ui/blob/master/libs/search/src/lib/sort-by/sort-by.component.ts#L19 ;).

2 things about translations:
* We should clean it up with proper key scheme before it is getting too large https://github.com/geonetwork/geonetwork-ui/blob/master/apps/search/src/assets/i18n/en.json
* I'm still not 100% sure how this will be done at the end. Those type of values can't be really hardcoded in the app - search fields will change and may come from somewhere else (eg. https://github.com/geonetwork/core-geonetwork/blob/4.0.x/web-ui/src/main/resources/catalog/components/elasticsearch/directives/FacetDirective.js#L129 for facet in v4)

So I would dedicate some time to a proper i18n PR to make proposal for a key scheme, cleanup and how to handle translations (not only those for buttons). Something to put on the agenda for the sprint next week. Ok for you?

Reply:
Added to the agenda https://github.com/geonetwork/core-geonetwork/wiki/GeoNetwork-UI-and-microservices-codesprint-December-2020